src/Input.js

```diff
@@ -1,28 +1,45 @@
 import React from 'react';
-import ReactDOM from 'react-dom';
 import constants from './constants';
 import cx from 'classnames';
 import uuid from 'node-uuid';
 
 class Input extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {hasValue: false};
+  }
   componentDidMount() {
-    if (this.props.type === 'select' && !this.props.browserDefault && typeof $ !== 'undefined') {
-      $(ReactDOM.findDOMNode(this.refs.inputEl)).material_select();
-    }
+    this._setupEl();
   }
-
   componentDidUpdate() {
-    if (this.props.type === 'select' && !this.props.browserDefault && typeof $ !== 'undefined') {
-      var $el = $(ReactDOM.findDOMNode(this.refs.inputEl));
-      $el.material_select('destroy');
-      $el.material_select();
+    this._setupEl();
+  }
+
+  _setupEl() {
+    if (typeof $ !== 'undefined') {
```

Comment:
@isaiah in this case it's not on change though, it's only when the component is mounted, or if it receives new props. otherwise the `active` state is already managed by the underlying `materialize.js`.

Moreover in `_setupEl` I am only calling `setState` if there is a change is the `hasValue` attribute. This can happen at most once for `componentDidMount`, and once per new props received.

Thanks for the quick feedback :)